File: cpp/55_Jump Game.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        
        if(nums.size()<2)
            return true;
        int i,j,k = nums[0];
        if(k==0) return false;
        for(i=1;i<nums.size()-1;i++)
        {
            k--;
            k = max(k,nums[i]);
            if(k==0) return false;
            
        }
        return true;
    }
};

File: java/55. Jump Game.java
class Solution {
    public boolean canJump(int[] nums) {
        int n = nums.length;
        int target = n-1;
        
        for(int i=n-2; i>=0; i--){
            if(nums[i] + i >= target){
                target = i;
            }
        }
        
        if(target == 0) return true;
        return false;
        
    }
}
